### ISSUE TITLE
Neovide closes immediately after starting in arch linux

### ISSUE THREAD

**Issue by skyveil** [neovide/neovide#1307]
**Describe the bug**
I installed neovide-git from AUR and on opening, it appears for a second and then closes without any warnings or errors 

**To Reproduce**
1. Install neovide-git from AUR
2. launch neovide

**Expected behavior**
Neovide opens without any issue

**Screenshots**
![output](https://user-images.githubusercontent.com/72389866/167353173-c25251a1-f33a-4eff-96a0-be7a74d49eaa.gif)

**Desktop (please complete the following information):**
 - OS: Arch Linux 5.17.5
 - Neovide Version 0.9.0
 - Neovim Version 0.7.0

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**
[neovide_rCURRENT.log](https://github.com/neovide/neovide/files/8649344/neovide_rCURRENT.log)

**Additional context**
**neovide** package from AUR runs as expected

**Comment by brmejia** [comment]
Same behaviour for me with the following error:

thread 'main' panicked at 'attempt to divide by zero', src/window/mod.rs:207:20

Desktop: 
    OS: Fedora Desktop 35
    Neovide Version 0.9.0 (Compiled from source using cargo)
    Neovim Version 0.7.0

![image](https://user-images.githubusercontent.com/1476904/167371240-27d615b0-d2c5-4de3-a9de-c3164b5f72f3.png)

**Comment by MultisampledNight** [comment]
Could you both please post your desktop environment, display protocol and display drivers?

**Comment by skyveil** [comment]
window manager: i3
display server: X
```
$ lspci -k
...
00:02.0 VGA compatible controller [0300]: Intel Corporation GeminiLake [UHD Graphics 605] [8086:3184] (rev 03)
        Subsystem: Lenovo Device [17aa:398a]
        Kernel driver in use: i915
        Kernel modules: i915
...
```

**Comment by brmejia** [comment]
Window manager: gnome 41
Display protocol: X
Drivers (Failing with both graphic cards): 
```console
$ lspci -k | rg -A 3 NVIDIA
01:00.0 3D controller: NVIDIA Corporation TU117M [GeForce GTX 1650 Mobile / Max-Q] (rev a1)
        Subsystem: Hewlett-Packard Company Device 8601
        Kernel driver in use: nvidia
        Kernel modules: nouveau, nvidia_drm, nvidia
$ lspci -k | rg -A 4 VGA
00:02.0 VGA compatible controller: Intel Corporation CoffeeLake-H GT2 [UHD Graphics 630]
        DeviceName:  Onboard IGD
        Subsystem: Dell Device 0905
        Kernel driver in use: i915
        Kernel modules: i915
```

**Comment by skyveil** [comment]
Any updates on this? It still seems to be broken.

**Comment by MultisampledNight** [comment]
(to be honest, I plainly forgot about this issue)

...the fix may be trivially easy, looking at

https://github.com/neovide/neovide/blob/7a3bf522edcdf24b87b082d800177b8099a35846/src/window/mod.rs#L205-L209

And indeed I can reproduce this by setting my scale to something below `1.0`, but strangely only when running X11.

Anyways, a fix for this might be by just changing those lines to

```rust
        let scale_factor = window.scale_factor();
        let new_size_scaled = PhysicalSize {
            width: (f64::from(new_size.width) / scale_factor) as u32,
            height: (f64::from(new_size.height) / scale_factor) as u32,
        };
```

But that may bring back what was fixed in #1216. @rikiwarren sorry to mention you again, but could you test [the `fix-scale-below-zero` on my fork](https://github.com/MultisampledNight/neovide/tree/fix-scale-below-zero) and see whether it breaks certain command line args again?

**Comment by TheMagzuz** [comment]
A temporary fix to this would be to prefix the neovide command `WINIT_X11_SCALE_FACTOR=1`, forcing the scaling factor to be 1.
